### ISSUE TITLE
[rllib] shorten jenkins tests

### ISSUE THREAD

**Issue by richardliaw** [ray-project/ray#5131]
Right now, there are 4 tests that run more than 5 minutes, and cause tune tests not to run and other rllib tests to get cut off:

```
++ docker run --rm --shm-size=20G --memory=20G c3255dfa34114acdd63dba1c924c94d51e806225d57aff4bf5595abed97497e8 /ray/ci/suppress_output python /ray/python/ray/rllib/tests/test_checkpoint_restore.py
WARNING: Your kernel does not support swap limit capabilities or the cgroup is not mounted. Memory limited without swap.
This command has been running for more than 5 minutes...
This command has been running for more than 10 minutes...

real	11m52.205s
user	12m55.307s
sys	1m53.243s
```
```
++ docker run --rm --shm-size=20G --memory=20G c3255dfa34114acdd63dba1c924c94d51e806225d57aff4bf5595abed97497e8 /ray/ci/suppress_output python /ray/python/ray/rllib/tests/test_supported_spaces.py
WARNING: Your kernel does not support swap limit capabilities or the cgroup is not mounted. Memory limited without swap.
This command has been running for more than 5 minutes...
This command has been running for more than 10 minutes...
This command has been running for more than 15 minutes...
This command has been running for more than 20 minutes...
This command has been running for more than 25 minutes...

real	26m58.102s
user	33m54.531s
sys	3m47.086s
```
```
++ for yaml in '$(ls $ROOT_DIR/../../python/ray/rllib/tuned_examples/regression_tests)'
++ docker run --rm --shm-size=20G --memory=20G c3255dfa34114acdd63dba1c924c94d51e806225d57aff4bf5595abed97497e8 /ray/ci/suppress_output python /ray/python/ray/rllib/tests/run_regression_tests.py /ray/python/ray/rllib/tuned_examples/regression_tests/pendulum-ddpg.yaml
WARNING: Your kernel does not support swap limit capabilities or the cgroup is not mounted. Memory limited without swap.
This command has been running for more than 5 minutes...
This command has been running for more than 10 minutes...

real	13m59.882s
user	0m56.641s
sys	0m23.592s
```
```
++ for yaml in '$(ls $ROOT_DIR/../../python/ray/rllib/tuned_examples/regression_tests)'
++ docker run --rm --shm-size=20G --memory=20G c3255dfa34114acdd63dba1c924c94d51e806225d57aff4bf5595abed97497e8 /ray/ci/suppress_output python /ray/python/ray/rllib/tests/run_regression_tests.py /ray/python/ray/rllib/tuned_examples/regression_tests/pendulum-ppo.yaml
WARNING: Your kernel does not support swap limit capabilities or the cgroup is not mounted. Memory limited without swap.
This command has been running for more than 5 minutes...

real	5m56.621s
user	0m34.542s
sys	0m13.875s
```

cc @ericl

**Comment by ericl** [comment]
I made a change in another PR that should cut down supported spaces by a
lot. Not sure we can trim the others though. What about bumping the Jenkins
timeout?

**Comment by simon-mo** [comment]
You can also run these docker tests in parallel using a make file then `make -j 4`. Jenkins machine are beefy enough to handle this level of concurrency. 

Alternative with simple shell job control
```
docker run ... &
docker run ... &
docker run ... &
docker run ... &
wait 
```
